Crashes on load

Dear developer,

I wanted to try this program as it looks pretty useful. I just wanted to load an existing anki set to be able to play around without having to create cards first.

Donkey failed on all sets I found online, e.g. https://github.com/PascalDierich/Rechnerarchitektur-anki-flashcards/blob/master/Computer%20Architecture.apkg

```
donkey load ~/Downloads/Computer\ Architecture.apkg
2024/06/22 21:06:48 hello
2024/06/22 21:06:48 hello
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/takacs/donkey/cmd.loadFileToDb({0x7fffad7c42cd?, 0x78f85f?}, {0x790476, 0x7}, 0xc0001180b8)
	/home/martin/build/go/pkg/mod/github.com/takacs/donkey@v0.0.0-20240622025837-e6d7f6e2d7aa/cmd/load.go:61 +0x42e
github.com/takacs/donkey/cmd.init.func5(0x9e1680, {0xc000126320, 0x1, 0x78f86f?})
	/home/martin/build/go/pkg/mod/github.com/takacs/donkey@v0.0.0-20240622025837-e6d7f6e2d7aa/cmd/load.go:40 +0xcb
github.com/spf13/cobra.(*Command).execute(0x9e1680, {0xc0001262e0, 0x1, 0x1})
	/home/martin/build/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:940 +0x882
github.com/spf13/cobra.(*Command).ExecuteC(0x9e1960)
	/home/martin/build/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
	/home/martin/build/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
github.com/takacs/donkey/cmd.Execute()
	/home/martin/build/go/pkg/mod/github.com/takacs/donkey@v0.0.0-20240622025837-e6d7f6e2d7aa/cmd/root.go:19 +0x1a
main.main()
	/home/martin/build/go/pkg/mod/github.com/takacs/donkey@v0.0.0-20240622025837-e6d7f6e2d7aa/main.go:9 +0xf
```

Version v0.0.0-20240622025837-e6d7f6e2d7aa

--- a/cmd/load.go
+++ b/cmd/load.go
@@ -68,7 +68,7 @@ func loadFileToDb(path string, deck string, cdb *card.CardDb) error {
 		fields := strings.Split(line, "\t")
 		cardId, err := cdb.Insert(fields[0], fields[1], deck)
 		if err != nil {
-			fmt.Printf("failed importing %v | %v\n", fields[0], fields[1])
+			fmt.Printf("failed importing %v | %v", fields[0], fields[1])
 		} else {
 			fmt.Printf("imported card front: %v | back: %v", fields[0], fields[1])
 		}

--- a/cmd/load.go
+++ b/cmd/load.go
@@ -2,8 +2,8 @@ package cmd
 
 import (
 	"bufio"
+	"errors"
 	"fmt"
-	"log"
 	"os"
 	"path/filepath"
 	"strings"
@@ -37,36 +37,50 @@ var loadCmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
-		loadFileToDb(args[0], deck, carddb)
+		err = loadFileToDb(args[0], deck, carddb)
+		if err != nil {
+			return err
+		}
 		return nil
 	},
 }
 
-func loadFileToDb(path string, deck string, cdb *card.CardDb) {
+func loadFileToDb(path string, deck string, cdb *card.CardDb) error {
+	if strings.HasSuffix(path, ".apkg") {
+		return errors.New(`.apkg format is not supported as of now. please export deck with Export Format -> Cards in Plain Text and uncheck Include HTML and media references from Anki and try again.`)
+	}
 	file, err := os.Open(filepath.Clean(path))
 	if err != nil {
-		log.Fatalf("failed to open file: %s", err)
+		return err
 	}
 	defer file.Close()
 
 	scanner := bufio.NewScanner(file)
 
-	var inserted int
+	inserted := 1
 	for scanner.Scan() {
 		line := scanner.Text()
+		fmt.Println(line)
 		if len(line) < 1 || line[0] == '#' {
+			fmt.Printf("couldn't import line %v\n", inserted)
 			continue
 		}
 		fields := strings.Split(line, "\t")
 		cardId, err := cdb.Insert(fields[0], fields[1], deck)
 		if err != nil {
-			log.Printf("failed importing %v | %v", fields[0], fields[1])
+			fmt.Printf("failed importing %v | %v\n", fields[0], fields[1])
+		} else {
+			fmt.Printf("imported card front: %v | back: %v", fields[0], fields[1])
 		}
 		supermemoDb, err := supermemo.New()
 		if err != nil {
-			log.Fatal(err)
+			fmt.Printf("couldn't import line %v\n", inserted)
 		}
-		supermemoDb.Insert(cardId)
+		err = supermemoDb.Insert(cardId)
+		if err != nil {
+			fmt.Printf("couldn't import line %v\n", inserted)
+			continue
+		}
 
 		inserted++
 	}
@@ -74,6 +88,7 @@ func loadFileToDb(path string, deck string, cdb *card.CardDb) {
 	fmt.Printf("inserted %v cards\n", inserted)
 
 	if err := scanner.Err(); err != nil {
-		log.Fatalf("error reading file: %s", err)
+		return err
 	}
+	return err
 }

--- a/cmd/load.go
+++ b/cmd/load.go
@@ -47,7 +47,7 @@ var loadCmd = &cobra.Command{
 
 func loadFileToDb(path string, deck string, cdb *card.CardDb) error {
 	if strings.HasSuffix(path, ".apkg") {
-		return errors.New(`.apkg format is not supported as of now. please export deck with Export Format -> Cards in Plain Text and uncheck Include HTML and media references from Anki and try again.`)
+		return errors.New(`.apkg format is not supported as of now. please export deck with Export Format -> Cards in Plain Text and uncheck Include HTML and media references from Anki and try again`)
 	}
 	file, err := os.Open(filepath.Clean(path))
 	if err != nil {

--- a/docs/load_cards.md
+++ b/docs/load_cards.md
@@ -5,7 +5,7 @@ Load functionality is currently supporting only one type of Anki export.
 To copy your deck from Anki export with the following steps:
 
 Click on the ⚙️ next to the deck you want to transfer and select `Export`
-<img width="664" alt="Screenshot 2024-06-22 at 19 30 32" src="https://github.com/takacs/donkey/assets/44911031/37387b06-367b-4282-a27b-7504aed7ddc8">
+<img width="656" alt="Screenshot 2024-06-22 at 19 59 08" src="https://github.com/takacs/donkey/assets/44911031/8f33c7b7-2235-436e-b77c-28312952c4be">
 
 Use Export format `Cards in Plain Text (.txt)` and uncheck `Include HTML and media references`.
 <img width="661" alt="Screenshot 2024-06-22 at 19 31 01" src="https://github.com/takacs/donkey/assets/44911031/cc2d79a8-d43e-4404-8c91-aea28cae36e1">

--- a/docs/load_cards.md
+++ b/docs/load_cards.md
@@ -1,0 +1,16 @@
+## How to load cards from Anki Export
+
+Load functionality is currently supporting only one type of Anki export.
+
+To copy your deck from Anki export with the following steps:
+
+Click on the ⚙️ next to the deck you want to transfer and select `Export`
+<img width="664" alt="Screenshot 2024-06-22 at 19 30 32" src="https://github.com/takacs/donkey/assets/44911031/37387b06-367b-4282-a27b-7504aed7ddc8">
+
+Use Export format `Cards in Plain Text (.txt)` and uncheck `Include HTML and media references`.
+<img width="661" alt="Screenshot 2024-06-22 at 19 31 01" src="https://github.com/takacs/donkey/assets/44911031/cc2d79a8-d43e-4404-8c91-aea28cae36e1">
+
+Once exported run the command
+```
+donkey load <path_to_export>
+```

--- a/internal/supermemo/supermemo.go
+++ b/internal/supermemo/supermemo.go
@@ -26,7 +26,7 @@ type SupermemoDb struct {
 	db *sql.DB
 }
 
-func (c *SupermemoDb) Insert(cardId uint) {
+func (c *SupermemoDb) Insert(cardId uint) error {
 	_, err := c.db.Exec(
 		"INSERT INTO supermemo(card_id, repetition, easiness_factor, interval, next_review_time) VALUES( ?, ?, ?, ?, ?)",
 		cardId,
@@ -36,9 +36,9 @@ func (c *SupermemoDb) Insert(cardId uint) {
 		time.Time{},
 	)
 	if err != nil {
-		log.Fatal(err)
+		return err
 	}
-	log.Print("hello")
+	return nil
 }
 
 func (c *SupermemoDb) Delete(id uint) error {
@@ -73,7 +73,10 @@ func (c *SupermemoDb) GetCardsSupermemo(cardId uint) Supermemo {
 
 	if !rows.Next() {
 		log.Printf("inserting card id %v", cardId)
-		c.Insert(cardId)
+		err := c.Insert(cardId)
+		if err != nil {
+			log.Fatal(err)
+		}
 		rows, err = c.db.Query(query)
 		if err != nil {
 			log.Fatal("query error supermemo", cardId)

--- a/tui/add_card.go
+++ b/tui/add_card.go
@@ -125,7 +125,10 @@ func (m *AddCardModel) submitCard() string {
 		fmt.Println("couldn't open db")
 	}
 	defer supermemodb.Close()
-	supermemodb.Insert(cardId)
+	err = supermemodb.Insert(cardId)
+	if err != nil {
+		log.Fatal(err)
+	}
 
 	m.inputs = defaultInputs(0)
 	return "Inserted!"

